code/game/machinery/atm.dm

```diff
@@ -0,0 +1,122 @@
+/obj/machinery/atm
+	name = "automated transfer machine"
+	desc = "A Nanotrasen Bank of Credit ATM, used for trasnferring funds to and from ID cards."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "dorm_available"
+	layer = 3
+	anchored = 1
+	var/obj/item/weapon/card/id/id = null //ID currently in the ATM
+	var/authed = 0 //If the ATM is authenticated
+
+/obj/machinery/atm/examine(mob/user)
+	..()
+	if(emagged)
+		user << "It appears to be in maintenance mode."
+
+/obj/machinery/atm/Destroy()
+	EjectId()
+	..()
+
+/obj/machinery/atm/emag_act(mob/user)
+	emagged = !emagged
+	audible_message("<span class='warning'>BZZZZZZzzzzzzz</span>")
+	user << "<span class='warning'>You [emagged ? "disable" : "enable"] [src]'s PIN authorization protocols.</span>"
+
+/obj/machinery/atm/power_change()
+	if(stat & BROKEN)
+		return
+	else
+		if(powered())
+			stat &= ~NOPOWER
+		else
+			icon_state = "dorm_off"
+			stat |= NOPOWER
+			EjectId() //If we run out of power, eject an ID if we have one
```

Comment:
You have no power to eject an ID? You could do the same as the hop console does, require an simple screwdriver to unscrew and remove the ID